equinox/nn/_embedding.py

```diff
@@ -89,3 +93,253 @@ def __call__(
                 "`eqx.nn.Embedding()(x)` should be called with a scalar index `x`. "
                 "Use `jax.vmap` if you would like to index with multiple values."
             )
+
+
+class RotaryPositionalEmbedding(Module):
+    embedding_size: int = field(static=True)
+    max_seq_len: Optional[int] = field(static=True)
+    freqs_cis: Optional[Complex[Array, "embedding_size/2"]] = field(static=True)
```

Comment:
> Also, it shouldn't be static if it will be an array.

I always thought that everything not marked as `static=True` will be affected by the `eqx.filter_grad(...)` function and thus be changed when calling `optax.apply_updates`. In this case, we don't want those to change, right?

Reply:
> I think to make the annotation here work, it should be `"{self.embedding_size}/2"`.

The ruff linter gave me trouble since it couldn't find `self` there. I tried this `RotaryPositionalEmbedding.embedding_size / 2` and it didn't complain any more. WDYT?

Reply:
> > Also, it shouldn't be static if it will be an array.
> 
> I always thought that everything not marked as `static=True` will be affected by the `eqx.filter_grad(...)` function and thus be changed when calling `optax.apply_updates`. In this case, we don't want those to change, right?

It's been so long that I even forgot my own implementation, lol. I had this in place just for that case:

`freqs_cis = jax.lax.stop_gradient(freqs_cis)`

But for my own understanding: any array marked with `static=True` would - during `jax.grad()` behave the same as if I do `jax.lax.stop_gradient(...)`?

Reply:
On `static=True`: it's true that it won't be seen by gradients etc. However, you can basically never use arrays as static fields, due to this:
```python
import equinox as eqx
import jax
import jax.numpy as jnp

class Foo(eqx.Module):
    x: jnp.ndarray = eqx.field(static=True)

foo = Foo(jnp.arange(3))
bar = Foo(jnp.arange(3) + 1)

f = jax.jit(lambda x: x.x + 1)
f(foo)
f(bar)
```
what's going on here is that in the `f(bar)` call, JAX will try to compare static fields for equality, to see whether or not it needs to recompile. And `foo.x = bar.x` returns a nonscalar array, not a bool, so it crashes.

In any case the above should all just be an academic question, since as per below I'm recommending not making this a field at all.

On ruff: prepend a space, i.e. `" {self.embedding_size}/2"`. I think that should make it work.